calc/CalcReader.scala

```diff
@@ -10,21 +10,43 @@ import distcompiler.calc.tokens.*
 
 object CalcReader extends Reader:
   override lazy val wellformed = Wellformed:
-    Node.Top ::= repeated(choice(tokens.Number, tokens.Expression, tokens.LowPrecOp, tokens.HighPrecOp))
+    Node.Top ::= repeated(choice(Number, AddOp, SubOp, MulOp, DivOp))
     
     Number ::= Atom
-    LowPrecOp ::= Atom
-    HighPrecOp ::= Atom
+    AddOp ::= Atom
+    SubOp ::= Atom
+    MulOp ::= Atom
+    DivOp ::= Atom
 
-    Expression ::= fields(
-      choice(tokens.Number, tokens.Expression),
-      choice(tokens.LowPrecOp, tokens.HighPrecOp),
-      choice(tokens.Number, tokens.Expression),
+    Add ::= fields(
```

Comment:
This is actually unreachable here - you can define it in mulDivPass where it's needed. That's easier to read imo.